Nos/Service/ReportPublisher.swift

```diff
@@ -0,0 +1,130 @@
+import Foundation
+import Dependencies
+import Logger
+import CoreData
+
+enum Reportinator {
+    // swiftlint:disable:next force_unwrapping
+    static var publicKey = PublicKey(npub: "npub14h23jzlyvumks4rvrz6ktk36dxfyru8qdf679k7q8uvxv0gm0vnsyqe2sh")!
+}
+
+class ReportPublisher {
+    @Dependency(\.relayService) private var relayService
+    @Dependency(\.currentUser) private var currentUser
+    @Dependency(\.analytics) private var analytics
+    
+    /// Publishes a report for the given target and category.
+    func publishPublicReport(for target: ReportTarget, category: ReportCategory, context: NSManagedObjectContext) {
+        guard let keyPair = currentUser.keyPair else {
+            Log.error("Cannot publish report - No signed in user")
+            return
+        }
+        
+        let publicReport = createPublicReport(for: target, category: category, pubKey: keyPair.publicKeyHex)
+        
+        Task {
+            do {
+                try await relayService.publishToAll(event: publicReport, signingKey: keyPair, context: context)
+                analytics.reported(target)
+            } catch {
+                Log.error("Failed to publish report: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    func publishPrivateReport(for target: ReportTarget, category: ReportCategory, context: NSManagedObjectContext) {
+        switch target {
+        case .note(let note):
+            guard let keyPair = currentUser.keyPair else {
+                Log.error("Cannot publish report - No signed in user")
+                return
+            }
+            
+            guard let reportRequestDM = createReportRequestDM(note: note, category: category, keyPair: keyPair) else {
+                Log.error("Failed to create gift wrapped report request")
+                return
+            }
+            
+            Task {
+                do {
+                    try await relayService.publish(
+                        event: reportRequestDM,
+                        to: Relay.nosAddress,
+                        context: context
+                    )
+                    analytics.reported(target)
+                } catch {
+                    Log.error("Failed to publish report: \(error.localizedDescription)")
+                }
+            }
+        case .author:
+            Log.error("Private reports for people are no implemented yet")
+            return
+        }
+    }
+    
+    // MARK: Helpers
+    
+    internal struct ReportRequest {
```

Comment:
you mean have it as a new kind for this use case?

Reply:
yeah, or we could probably just use a kind 1984 and put the reported event in a custom event tag, since it already has space for `reporterPubkey` and `reporterText`.